### PR TITLE
Add schedule compliance checks to reports site

### PR DESCRIPTION
**Pull request by owades**
# Description

This change updates the reports site to replace the file-presence checks with GTFS Schedule compliance checks instead.

The check logic is simplified slightly for this purpose, will all checks being marked as either:
- Pass
- Fail
- Not applicable/data unavailable

I will need to think through the impact of this a bit more (with Laurie's help, surely), to see whether we need to break this into more categories or describe better.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

## How has this been tested?
So far I've just been spot checking on a few organizations. I expect we'll do more detailed QA prior to launching (hopefully this can be next week).

## Screenshots (optional)
This is AC Transit for January '23
Screen Shot 2023-05-24 at 5 13 52 PM

### CODE REVIEW

File: templates/report.html.jinja
Comment (lauriemerrell) [inline templates/report.html.jinja:288]: `California Transit Data Guidelines` text is repeated (can remove this instance before the link I think?)

File: reports/generate_reports_data.py
Comment (lauriemerrell) [reply reports/generate_reports_data.py:155]: If we want to go really galaxy brain, we could think about whether there are ways that we want to show something in the UI for places where the data has changed since the prior month

File: templates/report.html.jinja
Comment (edasmalchi) [reply templates/report.html.jinja:288]: also looks like "to" should be added after adherence

File: reports/generate_reports_data.py
Comment (edasmalchi) [reply reports/generate_reports_data.py:155]: let's spin the "UI highlights change" idea into a separate issue to be addressed later

Comment (lauriemerrell) [reply reports/generate_reports_data.py:155]: https://github.com/cal-itp/reports/issues/257

**Comment by owades** [comment]
Thank you @lauriemerrell, and you are an essential reviewer! I just hadn't gotten around to tagging folks beyond the auto-tag first. I agree with Eric's replies above and will follow through on the suggestion to explain check frequency.

> I think that this simplification sounds good, my only quibble is that I think it would be preferable to add this as a column in the actual mart table (maybe it could be called `reports_status`), just so that for troubleshooting purposes we have a very easy way to check what we think the status will be on the reports site. (One of the goals of the v2 rewrite for the reports site was to have minimal data-related logic in the reports generation code and try to have it all in the warehouse where it's easier to check, for example you can just see it all in Metabase in advance without having to run reports generation code.)

I totally agree, thanks for suggesting this. I created [this ticket](https://github.com/cal-itp/reports/issues/258) as a followup there. I still need to work out a dbt issue, hopefully will get it sorted out this week and can draft a PR. One nice thing about the existing setup is that we can technically make this change after going live, since the outputs of `fct_monthly_reports_site_organization_guideline_checks` don't need to change. It would be best to do this before going live, though.

**Comment by lauriemerrell** [comment]
That sounds good, thanks @owades. 

Just re: 
>  since the outputs of `fct_monthly_reports_site_organization_guideline_checks` don't need to change.
Technically I think it would be ideal if the reports generation code looked at `reports_status` once that column exists, but it's not a huge deal.

File: templates/report.html.jinja
Comment (owades) [reply templates/report.html.jinja]: thanks y'all

**Comment by owades** [comment]
@lauriemerrell, do you know/remember if we are tracking which checks are manual versus automated? A binary value of "is_manual" could control something like an astrix here.

In the case where we didn't have a note about the check frequency, a complaint from an organization about their status would trigger us re-checking them and updating the value. That doesn't seem horrible to me. @edasmalchi 

> To Laurie's point on the manual checks, I think a "date assessed" might make things a little cluttered but perhaps we could note that those checks are infrequent somehow (like with an asterisk and a note further down like "assessed manually about every n months, contact us with any questions")

**Comment by lauriemerrell** [comment]
@owades I don't think we officially have an `is_manual`, but it wouldn't be too hard to add. I think the easiest way to do so would be to add a `seed` file that lists the manual checks, and then we could join it in when needed. We could also just add it as a column in the reports site checks mart table and basically be like `check in (list of manual checks) AS is_infrequent`.

**Comment by edasmalchi** [comment]
> @owades I don't think we officially have an `is_manual`, but it wouldn't be too hard to add. I think the easiest way to do so would be to add a `seed` file that lists the manual checks, and then we could join it in when needed. We could also just add it as a column in the reports site checks mart table and basically be like `check in (list of manual checks) AS is_infrequent`.

That sounds good, do we think there's capacity to do that before merging this or should we hardcode into the report for now and revisit when that's in the warehouse?

**Comment by lauriemerrell** [comment]
Seems like it could make sense to hard-code it here, do a follow-up PR in `data-infra` with both that change and the `reports_status` change, and then make a cleanup PR back here to use the new functionality from `data-infra`. But defer to y'all, also may depend if @owades has capacity to do this himself or whether you'd want help from Data Services.

**Comment by owades** [comment]
Here's a screenshot of how it looks with the latest updates:
Screen Shot 2023-05-31 at 11 48 04 AM

**Comment by owades** [comment]
@edasmalchi @lauriemerrell @charlie-costanzo I've made the updates we discussed last week, would love another review from y'all!

Eric, I want to make sure you review the copy, I really defer to you on that.

**Comment by edasmalchi** [comment]
the copy (and everything else) looks good to me!

File: templates/report.html.jinja
Comment (lauriemerrell) [inline templates/report.html.jinja:300]: Should `let us know` have a link to an appropriate email address where people could report this? (I also defer to @edasmalchi on copy)

File: templates/report.html.jinja
Comment (lauriemerrell) [inline templates/report.html.jinja:344]: We could add a link to the validator here, for example https://gtfs-validator.mobilitydata.org/?

**Review by lauriemerrell — APPROVED** [review]
I am good to merge, just a few optional suggestions

File: reports/generate_reports_data.py
Comment (owades) [reply reports/generate_reports_data.py:157]: Thanks for this, I will make followup tickets!

File: templates/report.html.jinja
Comment (owades) [reply templates/report.html.jinja]: Done! I opted to link to https://gtfs.org/schedule/validate/, which has a bit more summary info, and then links to the actual validator tool

File: templates/report.html.jinja
Comment (owades) [reply templates/report.html.jinja]: done!

**Review by lauriemerrell — APPROVED** [review]

**Review by charlie-costanzo — CHANGES_REQUESTED** [review]
Hey Scott, thanks for taking care of this! Generating the data and building the site both seem to be working locally - my only request would be to update the schema template used by the test `python test_report_data.py` which checks output schema, [described here in the readme](https://github.com/cal-itp/reports#validating-the-report-creation). [The file is found here](https://github.com/cal-itp/reports/blob/main/tests/schemas/schema_4_file_check.json). Let me know if you're not able to do that before you go OOO or if you have any questions. Thanks again!

**Review by charlie-costanzo — APPROVED** [review]
Thank you for making these changes @owades! I got the tests to pass locally and things look great.

**Pull request merged**
